octahedroid/drupal-decoupled
# ✨ feat(create-drupal-decoupled) Add next scaffold

# What this PR does?
- Add next as supported FE framework
- Removed default to `--frontend` flag
- Moved files to shared folder, so they can handle unspecific frontend files
- Supported renaming for files
- Print generated files

# How to test?

- Ensure you have a local drupal
- In a terminal go where you want to have your Next project
- Execute
```bash
npx create-next-app@latest
```
- Select the default options
```
✔ What is your project named? … my-app
✔ Would you like to use TypeScript? … No / Yes
✔ Would you like to use ESLint? … No / Yes
✔ Would you like to use Tailwind CSS? … No / Yes
✔ Would you like to use `src/` directory? … No / Yes
✔ Would you like to use App Router? (recommended) … No / Yes
✔ Would you like to customize the default import alias (@/*)? … No / Yes
```
- now execute the scaffold command

```bash
npm create @octahedroid/drupal-decoupled@0.4.0-preview-acab7a3-20240801210633 -- --frontend next
```
- copy the .env.example to .env.local and fill it content with the Drupal config
```
cp .env.example .env.local
```
- install the needed dependencies
```
npm install urql graphql drupal-auth-client
```
- Run the service
```
npm run dev -- --port 8080
```
- Visit any node to view the content
